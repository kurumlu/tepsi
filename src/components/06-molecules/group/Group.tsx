import * as React from 'react';
import cn from 'classnames';
import { FunctionComponent, Ref } from 'react';

export enum GroupAlign {
  LEFT = 'left',
  CENTER = 'center',
  RIGHT = 'right',
  SPACE_AROUND = 'space-around',
  SPACE_BETWEEN = 'space-between',
}

export enum GroupVerticalAlign {
  TOP = 'top',
  CENTER = 'center',
  BOTTOM = 'bottom',
  BASELINE = 'baseline',
}

export enum GroupGutter {
  NONE = '',
  XSMALL = 'xs',
  SMALL = 's',
  MEDIUM = 'm',
  LARGE = 'l',
  XLARGE = 'xl',
}

export enum GroupWrap {
  NOWRAP = 'nowrap',
  WRAP = 'wrap',
  REVERSE = 'wrap-reverse',
}

export type GroupProps = {
  align?: GroupAlign;
  className?: string;
  column?: boolean;
  dataAnalyticsBannerid?: string;
  dataQA?: string;
  equalSize?: boolean;
  gutter?: GroupGutter;
  gutterX?: GroupGutter;
  gutterY?: GroupGutter;
  inline?: boolean;
  valign?: GroupVerticalAlign;
  fullHeight?: boolean;
  innerRef?: Ref<HTMLDivElement>;
  wrap?: GroupWrap;
};

export const Group: FunctionComponent<GroupProps> = ({
  align = GroupAlign.LEFT,
  children,
  className,
  column,
  dataAnalyticsBannerid,
  dataQA,
  equalSize,
  gutter = GroupGutter.XSMALL,
  gutterX,
  gutterY,
  inline,
  valign,
  fullHeight,
  innerRef,
  wrap,
}) => (
  <div
    ref={innerRef}
    data-qa={dataQA}
    data-analytics-bannerid={dataAnalyticsBannerid}
    className={cn(
      'sd-m-group',
      {
        [`sd-m-group--${gutter}`]: gutterX || gutterY ? null : gutter,
        [`sd-m-group--x-${gutterX}`]: gutterX,
        [`sd-m-group--y-${gutterY}`]: gutterY,
        'sd-m-group--inline': inline,
        'sd-m-group--equal-size': equalSize,
        [`sd-m-group--align-${align}`]: align,
        [`sd-m-group--valign-${valign}`]: valign,
        [`sd-m-group--wrap-${wrap}`]: wrap,
        'sd-m-group--column': column,
        'sd-m-group--full-height': fullHeight,
      },
      className
    )}>
    {children}
  </div>
);
