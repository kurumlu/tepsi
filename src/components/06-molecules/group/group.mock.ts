import {
  GroupProps,
  GroupGutter,
  GroupAlign,
  GroupVerticalAlign,
} from './Group';

export const mockGroupProps: GroupProps = {
  className: 'as-m-grout-test',
  valign: GroupVerticalAlign.CENTER,
  align: GroupAlign.CENTER,
  gutter: GroupGutter.MEDIUM,
  inline: true,
  dataQA: 'dataQA',
  dataAnalyticsBannerid: 'dataAnalyticsBannerid',
  fullHeight: true,
};
