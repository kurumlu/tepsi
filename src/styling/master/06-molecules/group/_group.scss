// Variables for Group MOLECULE
$m-group--xs--margin: get-spacing(.25) !default;
$m-group--s--margin:  get-spacing(.5)  !default;
$m-group--m--margin:  get-spacing(1)   !default;
$m-group--l--margin:  get-spacing(1.5) !default;
$m-group--xl--margin: get-spacing(2)   !default;


// Code for Group MOLECULE
.sd-m-group {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

// --align
.sd-m-group--align-left {
  justify-content: flex-start;
}

.sd-m-group--align-center {
  justify-content: center;
}

.sd-m-group--align-right {
  justify-content: flex-end;
}

.sd-m-group--align-space-around {
  justify-content: space-around;
}

.sd-m-group--align-space-between {
  justify-content: space-between;
}

.sd-m-group--valign-center {
  align-items: center;
}

.sd-m-group--valign-top {
  align-items: flex-start;
}

.sd-m-group--valign-bottom {
  align-items: flex-end;
}

.sd-m-group--valign-baseline {
  align-items: baseline;
}

.sd-m-group--inline {
  flex-wrap: nowrap;
}

.sd-m-group--equal-size > * {
  flex: 1;
}

.sd-m-group--full-height {
  height: 100%;
}

// --column
.sd-m-group--column {
  flex-direction: column;
  flex-wrap: nowrap;
}

// --gutter
$m-group-gutter-map: (
  xs: $m-group--xs--margin,
  s: $m-group--s--margin,
  m: $m-group--m--margin,
  l: $m-group--l--margin,
  xl: $m-group--xl--margin,
);

// Generate code for Group gutters
@each $size, $gutter in $m-group-gutter-map {
  .sd-m-group--#{$size} {
    margin: -1 * $gutter;

    > * {
      margin: $gutter;
    }
  }

  .sd-m-group--x-#{$size} {
    margin-left: -1 * $gutter;
    margin-right: -1 * $gutter;

    > * {
      margin-left: $gutter;
      margin-right: $gutter;
    }
  }

  .sd-m-group--y-#{$size} {
    margin-top: -1 * $gutter;
    margin-bottom: -1 * $gutter;

    > * {
      margin-top: $gutter;
      margin-bottom: $gutter;
    }
  }
}

// https://css-tricks.com/flexbox-truncated-text/
.sd-m-group--inline .sd-t-box {
  min-width: 0;
}

// --wrap
.sd-m-group--wrap-wrap {
  flex-wrap: wrap;
}
.sd-m-group--wrap-nowrap {
  flex-wrap: nowrap;
}
.sd-m-group--wrap-wrap-reverse {
  flex-wrap: wrap-reverse;
}