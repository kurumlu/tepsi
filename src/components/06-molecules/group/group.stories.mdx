import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs';
import { Group, GroupAlign, GroupVerticalAlign, GroupGutter, GroupWrap } from './Group';
import { Tag, TagColor } from '../../05-atoms/tag/Tag';
import { Badge } from '../../05-atoms/badge/Badge';

<Meta title="Design system/Patterns/Molecules/Group" />

# Group

Use to group a bunch of items together and make them wrap responsively.

## Available props:
<ArgsTable of={Group} />

#### GroupAlign
| Key             |
|-----------------|
| `LEFT`          |
| `CENTER`        |
| `RIGHT`         |
| `SPACE_BETWEEN` |

#### GroupVerticalAlign
| Key      |
|----------|
| `TOP`    |
| `CENTER` |
| `BOTTOM` |
| `BASELINE` |

#### GroupGutter
| Key      |
|----------|
| `NONE`   |
| `XSMALL` |
| `SMALL`  |
| `MEDIUM` |
| `LARGE`  |
| `XLARGE` |

#### GroupWrap
| Key       |
|-----------|
| `WRAP`    |
| `NOWRAP`  |
| `REVERSE` |

## Example

### Default
<Canvas>
  <Story name="Default">
    <Group>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"></Badge>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"></Badge>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"></Badge>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

### Align

Choose the way the children are aligned.

#### Left
<Canvas>
  <Story name="Align - Left">
    <Group align={GroupAlign.LEFT}>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"></Badge>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"></Badge>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"></Badge>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"></Badge>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"></Badge>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"></Badge>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

#### Center
<Canvas>
  <Story name="Align - Center">
    <Group align={GroupAlign.CENTER}>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

#### Right
<Canvas>
  <Story name="Align - Right">
    <Group align={GroupAlign.RIGHT}>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

#### Space around
<Canvas>
  <Story name="Align - Space Around">
    <Group align={GroupAlign.SPACE_AROUND}>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

#### Space between
<Canvas>
  <Story name="Align - Space Between">
    <Group align={GroupAlign.SPACE_BETWEEN}>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>


### Gutter

Choose a gutter size.

#### Extra small
<Canvas>
  <Story name="Gutter - Extra small">
    <Group gutter={GroupGutter.XSMALL}>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

#### Small
<Canvas>
  <Story name="Gutter - Small">
    <Group gutter={GroupGutter.SMALL}>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

#### Medium
<Canvas>
  <Story name="Gutter - Medium">
    <Group gutter={GroupGutter.MEDIUM}>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

#### Large
<Canvas>
  <Story name="Gutter - Large">
    <Group gutter={GroupGutter.LARGE}>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

#### Extra large
<Canvas>
  <Story name="Gutter - Extra large">
    <Group gutter={GroupGutter.XLARGE}>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

#### None
<Canvas>
  <Story name="Gutter - None">
    <Group gutter={GroupGutter.NONE}>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

### Inline

Items are displayed in a single line.

<Canvas>
  <Story name="Inline">
    <Group align={GroupAlign.LEFT} inline>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

### Equal size

The children divide the full width of the group container equally among themselves.

> Please note, no need to use __inline__ prop in addition in this case.

<Canvas>
  <Story name="Inline uniform">
    <Group align={GroupAlign.LEFT} equalSize>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

### Column

Items are displayed in a single column.

<Canvas>
  <Story name="Column">
    <Group column>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>


### Wrap
<Canvas>
  <Story name="Wrap - nowrap">
    <Group wrap={GroupWrap.NOWRAP}>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>

<Canvas>
  <Story name="Wrap - reverse">
    <Group wrap={GroupWrap.REVERSE}>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
      <Tag>
        Pre-order
      </Tag>
      <Tag color={TagColor.SECONDARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.TERTIARY}>
        <Badge count="5"/>
      </Tag>
      <Tag color={TagColor.QUATERNARY}>
        <Badge count="5"/>
        &nbsp;
        Offer you can't refuse
      </Tag>
      <Tag color={TagColor.QUINARY}>
        Discount 50%
      </Tag>
    </Group>
  </Story>
</Canvas>