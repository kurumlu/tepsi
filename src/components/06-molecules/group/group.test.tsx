import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Group, GroupProps, GroupWrap } from './Group';
import { mockGroupProps } from './group.mock';

describe('<Group />', () => {
  it('renders with default props', () => {
    const component: ShallowWrapper<GroupProps> = shallow(<Group />);

    expect(component).toMatchSnapshot();
  });

  it('renders with props', () => {
    const component: ShallowWrapper<GroupProps> = shallow(
      <Group {...mockGroupProps} />
    );

    expect(component).toMatchSnapshot();
  });

  it('renders with children', () => {
    const component: ShallowWrapper<GroupProps> = shallow(
      <Group>
        <div>Children</div>
      </Group>
    );

    expect(component).toMatchSnapshot();
  });

  it('renders with column prop', () => {
    const component: ShallowWrapper<GroupProps> = shallow(<Group column />);

    expect(component.hasClass('sd-m-group--column')).toBe(true);
    expect(component).toMatchSnapshot('with column prop');
  });

  it('renders with equalSize prop', () => {
    const component: ShallowWrapper<GroupProps> = shallow(<Group equalSize />);

    expect(component.hasClass('sd-m-group--equal-size')).toBeTruthy();
    expect(component).toMatchSnapshot('with inlineUniform prop');
  });

  it('renders with wrap prop', () => {
    const component: ShallowWrapper<GroupProps> = shallow(
      <Group wrap={GroupWrap.REVERSE} />
    );

    expect(component.hasClass('sd-m-group--wrap-wrap-reverse')).toBe(true);
    expect(component).toMatchSnapshot('with wrap prop');
  });
});
