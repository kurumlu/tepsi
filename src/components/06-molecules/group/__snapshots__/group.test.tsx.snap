// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Group /> renders with children 1`] = `
<div
  className="sd-m-group sd-m-group--xs sd-m-group--align-left"
>
  <div>
    Children
  </div>
</div>
`;

exports[`<Group /> renders with column prop: with column prop 1`] = `
<div
  className="sd-m-group sd-m-group--xs sd-m-group--align-left sd-m-group--column"
/>
`;

exports[`<Group /> renders with default props 1`] = `
<div
  className="sd-m-group sd-m-group--xs sd-m-group--align-left"
/>
`;

exports[`<Group /> renders with equalSize prop: with inlineUniform prop 1`] = `
<div
  className="sd-m-group sd-m-group--xs sd-m-group--equal-size sd-m-group--align-left"
/>
`;

exports[`<Group /> renders with props 1`] = `
<div
  className="sd-m-group sd-m-group--m sd-m-group--inline sd-m-group--align-center sd-m-group--valign-center sd-m-group--full-height sd-m-grout-test"
  data-analytics-bannerid="dataAnalyticsBannerid"
  data-qa="dataQA"
/>
`;

exports[`<Group /> renders with wrap prop: with wrap prop 1`] = `
<div
  className="sd-m-group sd-m-group--xs sd-m-group--align-left sd-m-group--wrap-wrap-reverse"
/>
`;
